### ISSUE TITLE
Autodoc errors for documentation of demand/metamodel/ nn_generator

### ISSUE THREAD

**Issue by Jack-Hawthorne** [architecture-building-systems/CityEnergyAnalyst#1725]
Autodoc throws up the following error when documenting demand/metamodel/ nn_generator scripts:

```
c:\users\jack\documents\github\cityenergyanalyst\cea\demand\metamodel\nn_generator\input_matrix.py:docstring of cea.demand.metamodel.nn_generator.input_matrix.prep_NN_delay:3: WARNING: Unexpected indentation.
c:\users\jack\documents\github\cityenergyanalyst\cea\demand\metamodel\nn_generator\input_matrix.py:docstring of cea.demand.metamodel.nn_generator.input_matrix.prep_NN_delay:4: WARNING: Block quote ends without a blank line; unexpected unindent.
WARNING: autodoc: failed to import module u'nn_pretrainer_continue' from module u'cea.demand.metamodel.nn_generator'; the following exception was raised:
Traceback (most recent call last):
  File "C:\Users\Jack\Miniconda2\envs\cea\lib\site-packages\sphinx\ext\autodoc\importer.py", line 154, in import_module
    __import__(modname)
  File "c:\users\jack\documents\github\cityenergyanalyst\cea\demand\metamodel\nn_generator\nn_pretrainer_continue.py", line 21, in <module>
    from cea.demand.metamodel.nn_generator.nn_trainer_resume import neural_trainer_resume, nn_model_collector
  File "c:\users\jack\documents\github\cityenergyanalyst\cea\demand\metamodel\nn_generator\nn_trainer_resume.py", line 10, in <module>
    from keras.callbacks import EarlyStopping
RuntimeError: sys.path must be a list of directory names

WARNING: autodoc: failed to import module u'nn_pretrainer_pipeline' from module u'cea.demand.metamodel.nn_generator'; the following exception was raised:
Traceback (most recent call last):
  File "C:\Users\Jack\Miniconda2\envs\cea\lib\site-packages\sphinx\ext\autodoc\importer.py", line 154, in import_module
    __import__(modname)
  File "c:\users\jack\documents\github\cityenergyanalyst\cea\demand\metamodel\nn_generator\nn_pretrainer_pipeline.py", line 21, in <module>
    from cea.demand.metamodel.nn_generator.nn_trainer import neural_trainer
  File "c:\users\jack\documents\github\cityenergyanalyst\cea\demand\metamodel\nn_generator\nn_trainer.py", line 22, in <module>
    from keras.callbacks import EarlyStopping
RuntimeError: sys.path must be a list of directory names

WARNING: autodoc: failed to import module u'nn_random_pipeline' from module u'cea.demand.metamodel.nn_generator'; the following exception was raised:
Traceback (most recent call last):
  File "C:\Users\Jack\Miniconda2\envs\cea\lib\site-packages\sphinx\ext\autodoc\importer.py", line 154, in import_module
    __import__(modname)
  File "c:\users\jack\documents\github\cityenergyanalyst\cea\demand\metamodel\nn_generator\nn_random_pipeline.py", line 22, in <module>
    from cea.demand.metamodel.nn_generator.nn_trainer import neural_trainer, nn_input_collector
  File "c:\users\jack\documents\github\cityenergyanalyst\cea\demand\metamodel\nn_generator\nn_trainer.py", line 22, in <module>
    from keras.callbacks import EarlyStopping
RuntimeError: sys.path must be a list of directory names

WARNING: autodoc: failed to import module u'nn_trainer' from module u'cea.demand.metamodel.nn_generator'; the following exception was raised:
Traceback (most recent call last):
  File "C:\Users\Jack\Miniconda2\envs\cea\lib\site-packages\sphinx\ext\autodoc\importer.py", line 154, in import_module
    __import__(modname)
  File "c:\users\jack\documents\github\cityenergyanalyst\cea\demand\metamodel\nn_generator\nn_trainer.py", line 22, in <module>
    from keras.callbacks import EarlyStopping
RuntimeError: sys.path must be a list of directory names

WARNING: autodoc: failed to import module u'nn_trainer_estimate' from module u'cea.demand.metamodel.nn_generator'; the following exception was raised:
Traceback (most recent call last):
  File "C:\Users\Jack\Miniconda2\envs\cea\lib\site-packages\sphinx\ext\autodoc\importer.py", line 154, in import_module
    __import__(modname)
  File "c:\users\jack\documents\github\cityenergyanalyst\cea\demand\metamodel\nn_generator\nn_trainer_estimate.py", line 17, in <module>
    from cea.demand.metamodel.nn_generator.nn_trainer_resume import nn_model_collector
  File "c:\users\jack\documents\github\cityenergyanalyst\cea\demand\metamodel\nn_generator\nn_trainer_resume.py", line 10, in <module>
    from keras.callbacks import EarlyStopping
RuntimeError: sys.path must be a list of directory names

WARNING: autodoc: failed to import module u'nn_trainer_evaluate' from module u'cea.demand.metamodel.nn_generator'; the following exception was raised:
Traceback (most recent call last):
  File "C:\Users\Jack\Miniconda2\envs\cea\lib\site-packages\sphinx\ext\autodoc\importer.py", line 154, in import_module
    __import__(modname)
  File "c:\users\jack\documents\github\cityenergyanalyst\cea\demand\metamodel\nn_generator\nn_trainer_evaluate.py", line 14, in <module>
    from cea.demand.metamodel.nn_generator.nn_trainer_resume import nn_model_collector
  File "c:\users\jack\documents\github\cityenergyanalyst\cea\demand\metamodel\nn_generator\nn_trainer_resume.py", line 10, in <module>
    from keras.callbacks import EarlyStopping
RuntimeError: sys.path must be a list of directory names

WARNING: autodoc: failed to import module u'nn_trainer_resume' from module u'cea.demand.metamodel.nn_generator'; the following exception was raised:
Traceback (most recent call last):
  File "C:\Users\Jack\Miniconda2\envs\cea\lib\site-packages\sphinx\ext\autodoc\importer.py", line 154, in import_module
    __import__(modname)
  File "c:\users\jack\documents\github\cityenergyanalyst\cea\demand\metamodel\nn_generator\nn_trainer_resume.py", line 10, in <module>
    from keras.callbacks import EarlyStopping
RuntimeError: sys.path must be a list of directory names

WARNING: autodoc: failed to import module u'nn_trainer_resume_single' from module u'cea.demand.metamodel.nn_generator'; the following exception was raised:
No module named nn_generator.nn_settings
```

Seems like keras.callbacks has to be added to the mock modules in the docs/conf.py
along with some formatting errors with indentation etc...

**Comment by Jack-Hawthorne** [comment]
added keras.callbacks to MOCK_MODULES in the docs/config and get the following errors
``` c:\users\jack\documents\github\cityenergyanalyst\cea\demand\metamodel\nn_generator\input_matrix.py:docstring of cea.demand.metamodel.nn_generator.input_matrix.prep_NN_delay:3: WARNING: Unexpected indentation.
c:\users\jack\documents\github\cityenergyanalyst\cea\demand\metamodel\nn_generator\input_matrix.py:docstring of cea.demand.metamodel.nn_generator.input_matrix.prep_NN_delay:4: WARNING: Block quote ends without a blank line; unexpected unindent.
c:\users\jack\documents\github\cityenergyanalyst\cea\demand\metamodel\nn_generator\nn_trainer_estimate.py:docstring of cea.demand.metamodel.nn_generator.nn_trainer_estimate.prep_NN_delay_estimate:3: WARNING: Unexpected indentation.
c:\users\jack\documents\github\cityenergyanalyst\cea\demand\metamodel\nn_generator\nn_trainer_estimate.py:docstring of cea.demand.metamodel.nn_generator.nn_trainer_estimate.prep_NN_delay_estimate:4: WARNING: Block quote ends without a blank line; unexpected unindent.                                                                     WARNING: autodoc: failed to import module u'nn_trainer_resume_single' from module u'cea.demand.metamodel.nn_generator'; the following exception was raised:             No module named nn_generator.nn_settings
C:\Users\Jack\Documents\GitHub\CityEnergyAnalyst\docs\modules\cea.optimization.rst:12: WARNING: toctree contains reference to nonexisting document u'modules/cea.optimization.individual_evaluation'
c:\users\jack\documents\github\cityenergyanalyst\cea\optimization\optimization_main.py:docstring of cea.optimization.optimization_main.moo_optimization:3: WARNING: Unexpected indentation.
```

**Comment by daren-thomas** [comment]
@JIMENOFONSECA, @Jack-Hawthorne I believe the nn_generator stuff is not really functional anyway, right? What is the timeline for this? Or do we move it to a branch and forget about it for now?

**Comment by Jack-Hawthorne** [comment]
@daren-thomas I'm just going through the list of exclusions we have in the docs/conf to resolve (hopefully) all the sphinx build errors.

The timeline is end of year for all the build errors, i'm not addressing the functionality of the scripts though.
